### PR TITLE
OBPIH-823 Add NPM as a dependency management tool plus webpack

### PR DESCRIPTION
**Pull request by awalkowiak**
https://tickets.pih-emr.org/browse/OBPIH-823

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  "version": "1.0.0",
+  "description": "OpenBoxes Web App",
+  "main": "index.js",
+  "directories": {
+    "doc": "docs",
+    "lib": "lib",
+    "test": "test"
+  },
+  "scripts": {
+    "watch": "webpack --watch --colors --progress",
```
Comment (kkaczmarczyk) [inline package.json]: Let's add info about 'npm watch' to README

**Comment by jmiranda** [comment]
@kkaczmarczyk Great point. Do we need to run `npm watch` in a console alongside `grails run-app`?Or is there a way to integrate it into the `grails run-app` command so that it gets spawned automatically?

**Comment by pmuchowski** [comment]
@jmiranda the `npm run watch` is used to automatically build the React after any changes, so it needs to run in separate console. We could try to run `npm run bundle`, but I'm not sure if it can be integrated into the `grails run-app` task.

**Comment by kkaczmarczyk** [comment]
@jmiranda initially we looked into it, but seems like run-app is a built-in command from grails wrapper. Would you know anything more about it?

We were actually considering adding gradle to manage build cycle through it. I also saw some documentation across internet about it. Would you call that action worthwhile?

**Comment by jmiranda** [comment]
@kkaczmarczyk Ok cool. Yeah we can hook into Grails build events (i.e. eventRunAppStart) to do something like this (see https://grails.github.io/grails2-doc/1.3.9/guide/12.%20Plug-ins.html#12.6%20Hooking%20into%20Build%20Events). I'll look into whether we can spawn an `npm watch` process using `eventRunAppStart` within `scripts/Events.groovy` (https://github.com/openboxes/openboxes/blob/develop/scripts/_Events.groovy). This could get ugly though (zombie processes) if we don't clean up after ourselves so we'll need to be careful.

**Comment by kkaczmarczyk** [comment]
@jmiranda Interesting, I guess we could do that in different ticket, as this PR blocks some further commits from our side. Or do you feel we should do that right away?
Anyhow I approved the latest changes.

**Comment by jmiranda** [comment]
No that would be a separate ticket if we have time. I'll merge this now.

**Pull request merged**